mesonbuild/interpreter.py

```diff
@@ -3408,8 +3409,16 @@ def func_configure_file(self, node, args, kwargs):
             raise InterpreterException('@INPUT@ used as command argument, but no input file specified.')
         # Validate output
         output = kwargs['output']
+        ofile_rpath = os.path.join(self.subdir, output)
         if not isinstance(output, str):
             raise InterpreterException('Output file name must be a string')
+        if ofile_rpath in self.configure_file_outputs:
+            if len(self.subdir) > 0:
+                mlog.warning('Output file', mlog.bold(ofile_rpath), 'for configure_file overwritten. First time written in subdir', self.subdir, 'line', self.configure_file_outputs[ofile_rpath], 'now in line', self.current_lineno)
+            else:
+                mlog.warning('Output file', mlog.bold(ofile_rpath), 'for configure_file overwritten. First time written in line', self.configure_file_outputs[ofile_rpath], 'now in line', self.current_lineno)
```

Comment:
This error can be simpler in three ways:

1. Use `subdir/meson.build:linenumber` like other warnings
2. Message could be clearer: `Output file 'blah' for configure_file() at subdir/meson.build:linenumber overwrites configure_file() output at subdir2/meson.build:linenumber2`
3. `mlog.bold()` doesn't work in all terminals, so you need to both quote and bold it

Reply:
For (3), coincidentally, I posted https://github.com/mesonbuild/meson/pull/3837 yesterday.